commissions/views.py

```diff
@@ -1,3 +1,83 @@
-from django.shortcuts import render
+from rest_framework import generics, status
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.response import Response
 
-# Create your views here.
+from commissions.models import Commission
+from commissions.serializers import CommissionSerializer, CreateCommissionSerializer
+
+
+class ListCommissionsView(generics.ListAPIView):
+    """
+    Returns all available commission in the database, as well as the IDs of all users holding it
+    """
+    model = Commission
+    permission_classes = [IsAuthenticated]
+    serializer_class = CommissionSerializer
+    queryset = Commission.objects.all()
+
+    def get(self, request, *args, **kwargs):
+        serializer = self.get_serializer(self.get_queryset(), many=True)
+
+        return Response(serializer.data, status=status.HTTP_200_OK)
+
+
+class CreateCommissionView(generics.CreateAPIView):
+    """
+    Create a new commission in the database.
+    """
+    model = Commission
+    permission_classes = [IsAuthenticated]
+    serializer_class = CreateCommissionSerializer
+    queryset = Commission.objects.all()
+
+    def post(self, request, *args, **kwargs):
+        if 'name' not in request.data:
```

Comment:
I feel perhaps these two if statements should be handled by uniqueness/blankness in the Model (and then automatically by the Serializer)?

Reply:
You're (almost) right! This is handled by the serializer if you add a uniqueness validator to the serializer field.